tests/integration/components/registries/finalize-registration-modal/component-test.ts

```diff
@@ -102,4 +111,67 @@ module('Integration | Component | finalize-registration-modal', hooks => {
         // // Close the dialog
         this.set('isOpen', false);
     });
+
+    test('almost done modal content: no moderation', async function(assert) {
+        this.store = this.owner.lookup('service:store');
+        const noModerationProvider = server.create('registration-provider', { reviewsWorkflow: null });
+        const noModRegistration = server.create('registration', { provider: noModerationProvider });
+
+        const registrationModel = await this.store.findRecord('registration', noModRegistration.id);
+        this.set('draftManager', { provider: noModerationProvider });
+        this.set('model', registrationModel);
+        this.set('isOpen', true);
+        await render(hbs`
+            <Registries::FinalizeRegistrationModal::Manager
+                @registration={{this.model}}
+                @draftManager={{this.draftManager}}
+                as |manager|
+            >
+                <Registries::FinalizeRegistrationModal @isOpen={{this.isOpen}} @manager={{manager}} />
+            </Registries::FinalizeRegistrationModal::Manager>
+        `);
+        // Click immediate radio button
+        await click('[data-test-immediate-button]');
+        // Click submit button
+        await click('[data-test-submit-registration-button]');
+
+        const opts = { learnMoreLink: 'aaa.aa', htmlSafe: true };
+        assert.dom('[data-test-finalize-main]').hasTextContaining(
+            stripHtmlTags(t('registries.finalizeRegistrationModal.notice.noModerationFromProject', opts).toString()),
+            'modal shows warning',
+        );
+        assert.dom('[data-test-finalize-main]').doesNotHaveTextContaining(
+            'A moderator must review and approve', 'modal does not mention moderation for unmoderated providers',
+        );
+    });
+
+    test('almost done modal content: with moderation', async function(assert) {
+        this.store = this.owner.lookup('service:store');
+        const withModerationProvider = server.create('registration-provider');
+        const withModRegistration = server.create('registration', { provider: withModerationProvider });
+
+        const registrationModel = await this.store.findRecord('registration', withModRegistration.id);
+        this.set('draftManager', { provider: withModerationProvider, reviewsWorkflow: 'pre-moderation' });
```

Comment:
```suggestion
        this.set('draftManager', { provider: withModerationProvider });
```

Reply:
I had initially tried to run the tests like you're suggesting with just the provider defined, but the logic in the `finalize-registration-modal` evaluated `this.manager.draftManager.reviewsWorkflow` to be `undefined` in tests.
I am not too sure where exactly the problem lies, but I'm assuming that since `draftManager` here isn't an actual `DraftRegistrationManager`, some of the underlying logic may not be so reliable (ie: using an alias for a property on the provider).